benchmarks/time/src/main/scala/strawman/collection/immutable/ImmutableArrayBenchmark.scala

```diff
@@ -88,9 +113,38 @@ class ImmutableArrayBenchmark {
   def map(bh: Blackhole): Unit = bh.consume(xs.map(x => x + 1))
 
   @Benchmark
+  def reverse(bh: Blackhole): Any = bh.consume(xs.reverse)
```

Comment:
The benchmarks are inconsistent.  Some return Any, and some Unit.   This could affect the JVM optimizations.

Ideally the same benchmark suite is used for each of these as appropriate, only changing the underlying collection type each run.  `@Benchmark def reverse`, etc,  should not be written more than once.

Also, I am not sure why the form used here:

````scala
@Benchmark
def reverse(bs: Blackhole): Any = bh.consume(xs.reverse)
````

is preferred to 

````scala
@Benchmark
def reverse(): C = xs.reverse
````

(where `C` is the type of `xs`, though `Any` might do)

My understanding of JMH is that any value returned by the benchmark method is effectively blackhole'd anyway.  Obviously, in cases like foldLeft below we need the Blackhole to consume items that are not returned.

Many of the things above are not specific to this PR, but IMO may be affecting the sanity of the benchmark results.
Thoughts?